drivers/spi/Kconfig.stm32

```diff
@@ -24,6 +24,19 @@ config SPI_STM32_DMA
 	  Enable the SPI DMA mode for SPI instances
 	  that enable dma channels in their device tree node.
 
+config SPI_STM32_UNALIGNED_DMA
```

Comment:
Most of the time the SPI driver will be used by another driver, so users can't control if the buffer is aligned or not. I therefore believe this should be enabled by default.